From 12f0050b466a97e42113ff33e4868bf7906291d7 Mon Sep 17 00:00:00 2001
From: Stefan Haller <stefan@haller-berlin.de>
Date: Tue, 3 Nov 2020 17:16:31 +0100
Subject: [PATCH 14/18] git-gui: Auto-rescan on activate

Do an automatic rescan whenever the git-gui window receives focus. Most other
GUI tools do this, and it's very convenient; no more pressing F5 manually.

People who don't like this behavior can turn it off in the Options dialog.

Signed-off-by: Stefan Haller <stefan@haller-berlin.de>
Signed-off-by: Pratyush Yadav <me@yadavpratyush.com>
---
 git-gui/git-gui.sh     | 5 +++++
 git-gui/lib/option.tcl | 1 +
 2 files changed, 6 insertions(+)

diff --git a/git-gui/git-gui.sh b/git-gui/git-gui.sh
index 1983ac6d61..e6db9d970a 100755
--- a/git-gui/git-gui.sh
+++ b/git-gui/git-gui.sh
@@ -843,6 +843,7 @@ set font_descs {
 }
 set default_config(gui.stageuntracked) ask
 set default_config(gui.displayuntracked) true
+set default_config(gui.autorescan) true
 
 ######################################################################
 ##
@@ -3879,6 +3880,10 @@ if {[is_MacOSX]} {
 	bind .   <Alt-Key-4> {focus $::ui_comm}
 }
 
+if {[is_config_true gui.autorescan]} {
+	bind .   <FocusIn>  { if {"%W" eq "."} do_rescan }
+}
+
 set file_lists_last_clicked($ui_index) {}
 set file_lists_last_clicked($ui_workdir) {}
 
diff --git a/git-gui/lib/option.tcl b/git-gui/lib/option.tcl
index 487d70691d..cb2b3c1e88 100644
--- a/git-gui/lib/option.tcl
+++ b/git-gui/lib/option.tcl
@@ -145,6 +145,7 @@ proc do_options {} {
 		{b merge.diffstat {mc "Show Diffstat After Merge"}}
 		{t merge.tool {mc "Use Merge Tool"}}
 
+		{b gui.autorescan  {mc "Auto-Rescan On Activate"}}
 		{b gui.trustmtime  {mc "Trust File Modification Timestamps"}}
 		{b gui.pruneduringfetch {mc "Prune Tracking Branches During Fetch"}}
 		{b gui.matchtrackingbranch {mc "Match Tracking Branches"}}
-- 
2.51.2

